### DISCUSSION TITLE
如何使用wechaty来分享小程序？

### DISCUSSION THREAD

**Discussion by nzjcnjzx** [wechaty/wechaty#2085] [Q&A]
🚨 The issue tracker is not for questions 🚨

If you have a question, please ask it on https://stackoverflow.com/questions/tagged/wechaty

[question]

**Comment by nzjcnjzx** [comment]
*   if (/^link$/i.test(m.text())) {
   *     const miniProgramPayload = new MiniProgram ({
   *       username           : 'gh_xxxxxxx',     //get from mp.weixin.qq.com
   *       appid              : '',               //optional, get from mp.weixin.qq.com
   *       title              : '',               //optional
   *       pagepath           : '',               //optional
   *       description        : '',               //optional
   *       thumbnailurl       : '',               //optional
   *     })


thumbnailurl       这个参数的值应该填什么，怎么分享显示不出图片来?

**Comment by nzjcnjzx** [comment]
![image](https://user-images.githubusercontent.com/31735124/99085027-a978fa00-2602-11eb-9fb5-ad10f90c3da0.png)
上图中的key 值去哪里取？

**Comment by nzjcnjzx** [comment]
经过对比发现，上述的key的来源是已分享的小程序中获得的结果中的值

**Comment by nzjcnjzx** [comment]
![image](https://user-images.githubusercontent.com/31735124/99089193-f4494080-2607-11eb-84ad-455ff3442798.png)

**Comment by nzjcnjzx** [comment]
thumbnailurl 目前这个字段似乎没有作用， 现在通过上述6个字段已经可以进行小程序的分享了， 注意一点的是路径后面要加上.html 类似pages/bill/list/index.html这样

**Comment by Edward-Zhou** [comment]
@nzjcnjzx Could you share how you send the mini program? I am working on forwarding mini program, but fall into the issue that image did not show correctly.

Did you mean we just need to send out like below?
```
  username           : 'gh_xxxxxxx',     //get from mp.weixin.qq.com
  appid              : '',               //optional, get from mp.weixin.qq.com
  title              : '',               //optional
  pagepath           : '',               //optional
  description        : '',               //optional
  thumbkey       : '',               //optional
```

**Comment by aslanTT** [comment]
mee too. And can wechaty-puppet-wxwork send miniprogram? when I receive a miniprogram card message, but the message print out like:

`
receive message:{"_events":{},"_eventsCount":0,"id":"1002354","payload":{"filename":"","fromId":"7881299964965548","id":"1002354","mentionIdList":[],"roomId":"","text":"","timestamp":1606794859000,"toId":"1688850083602655","type":9}}
`
